extmod/machine_signal.c

```diff
@@ -41,20 +41,45 @@ typedef struct _machine_signal_t {
 } machine_signal_t;
 
 STATIC mp_obj_t signal_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    enum { ARG_pin, ARG_inverted };
-    static const mp_arg_t allowed_args[] = {
-        { MP_QSTR_, MP_ARG_OBJ | MP_ARG_REQUIRED },
-        { MP_QSTR_inverted, MP_ARG_BOOL, {.u_bool = false} },
-    };
+    mp_obj_t pin = args[0];
```

Comment:
Pin's are so fundamental, that id= is not supported for them. I believe we discussed that and agreed on that, and that's how esp8266 port (etc.) works. Otherwise arg num checking needs to be added, yeah.